### PR TITLE
Code cleanup: Follow-up of #13093

### PR DESCRIPTION
**Pull request by BlackbitDevs**
Follow-up as requested in https://github.com/pimcore/pimcore/pull/13093#issuecomment-1255906425

### CODE REVIEW

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 1]
```diff
@@ -628,6 +596,41 @@ public function getSelectOptions(Request $request)
         return new JsonResponse(['success' => true, 'options' => $options]);
     }
 
+    private function applyChanges(DataObject\Concrete $object, array $changes) {
```
Comment (blankse) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]:
```suggestion
    private function applyChanges(DataObject\Concrete $object, array $changes): void
    {
```

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 2]
```diff
@@ -1359,39 +1362,7 @@ public function saveAction(Request $request)
...
-                        $toDelete = $this->detectDeletedRemoteOwnerRelations($relations, $value);
-                        if (count($toAdd) > 0 || count($toDelete) > 0) {
-                            $this->processRemoteOwnerRelations($object, $toDelete, $toAdd, $fd->getOwnerFieldName());
-                        }
-                    } else {
-                        $object->setValue($key, $fd->getDataFromEditmode($value, $object, ['objectFromVersion' => $objectFromVersion]));
-                    }
-                }
-            }
+            $this->applyChanges($object, $this->decodeJson($request->get('changedData')));
```
Comment (mcop1) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]:
```suggestion
            $this->applyChanges($object, $this->decodeJson($request->get('data')));
```

**Review by mcop1 — COMMENTED** [review]
Tested selectOptionsProvider including object bricks and classification store, works fine, after I changed the suggested line of code.

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php (hunk 2 above)
Comment (mcop1) [reply bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: @BlackbitDevs any chance you had some time to look at the suggestion?

**Comment by BlackbitDevs** [comment]
@mcop1 Sorry, thought you had already committed it - but it was only a suggestion. Accepted it.

**Comment by mcop1** [comment]
> @mcop1 Sorry, thought you had already committed it - but it was only a suggestion. Accepted it.

No worries, thank you 👍

**Pull request merged**